packages/teleterm/src/ui/DocumentGateway/OfflineDocumentGateway.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import { ButtonPrimary, Flex, Text } from 'design';
+
+import * as Alerts from 'design/Alert';
+import Validation from 'shared/components/Validation';
+
+import LinearProgress from 'teleterm/ui/components/LinearProgress';
+
+import { PortFieldInput } from './common';
+import { DocumentGatewayProps } from './DocumentGateway';
+
+type OfflineDocumentGatewayProps = Pick<
+  DocumentGatewayProps,
+  'connectAttempt' | 'defaultPort' | 'reconnect'
+>;
+
+export function OfflineDocumentGateway(props: OfflineDocumentGatewayProps) {
+  const [port, setPort] = useState(props.defaultPort);
+  const statusDescription =
+    props.connectAttempt.status === 'processing' ? 'being set up' : 'offline';
+  const isProcessing = props.connectAttempt.status === 'processing';
+  const shouldShowPortInput =
+    props.connectAttempt.status === 'error' || port !== props.defaultPort;
+
+  return (
+    <Flex
+      maxWidth="590px"
+      width="100%"
+      flexDirection="column"
+      mx="auto"
+      alignItems="center"
+      mt={11}
+    >
+      <Text
+        typography="h5"
+        color="text.primary"
+        mb={2}
+        style={{ position: 'relative' }}
+      >
+        The database connection is {statusDescription}
+        {props.connectAttempt.status === 'processing' && <LinearProgress />}
+      </Text>
+      {props.connectAttempt.status === 'error' && (
+        <Alerts.Danger mb={0}>{props.connectAttempt.statusText}</Alerts.Danger>
+      )}
+      <Flex
+        as="form"
+        onSubmit={() => props.reconnect(port)}
+        alignItems="flex-end"
+        flexWrap="wrap"
+        justifyContent="space-between"
+        mt={3}
+        gap={2}
+      >
+        {shouldShowPortInput && (
+          <Validation>
+            <PortFieldInput
+              label="Port (optional)"
```

Comment:
Let's add `autoFocus` to this input.

Reply:
I'm not sure if we should do it here. I think autofocus makes sense in places where typing something is the only natural way to proceed, like the login form. Here creating a gateway might fail due to a variety of reasons (no network connection, lost access to the database server, expired certs). A port being occupied is only one of them so I don't think we should guide the user towards changing the port.

Is it okay if I leave it without autofocus then?

Reply:
Yeah, I'm convinced.